### PR TITLE
Upgrade tests

### PR DESCRIPTION
**Pull request by fao89**
[noissue]

### CODE REVIEW

File: plugin-template [hunk 1]
```diff
@@ -17,6 +17,7 @@ from jinja2 import Environment, FileSystemLoader
 
 DEFAULT_SETTINGS = {
     'additional_repos': [],
+    'upgrade_range': [],
```
Comment (dkliban) [inline plugin-template:20]: Please add this variable to the README

Comment (fao89) [reply plugin-template:20]: done!

File: templates/github/.github/workflows/ci.yml.j2 [hunk 2]
```diff
@@ -146,3 +146,89 @@ jobs:
...
+
+      - name: Install httpie
+        run: |
+          echo ::group::HTTPIE
+          sudo apt-get update -yq
+          sudo -E apt-get -yq --no-install-suggests --no-install-recommends install httpie
+          echo ::endgroup::
+          echo "TEST=upgrade" >> $GITHUB_ENV
+          echo "HTTPIE_CONFIG_DIR=$GITHUB_WORKSPACE/.ci/assets/httpie/" >> $GITHUB_ENV
+          {%- for plugin_branch, version in upgrade_range[0].items() %}
```
Comment (dkliban) [inline templates/github/.github/workflows/ci.yml.j2:188]: Why is the index 0 needed here?

Comment (fao89) [reply templates/github/.github/workflows/ci.yml.j2:188]: `upgrade_range` is a list of dicts:
```python
upgrade_range = [
   { "pulpcore_branch": "X.Y", "pulp_file_branch": "X.Y"},
   { "pulpcore_branch": "X.Y", "pulp_file_branch": "X.Y"}
]
```
the keys would repeat, so I took only the first dict keys

**Review by dkliban — APPROVED** [review]

**Pull request merged**